### ISSUE TITLE
Logging framework

### ISSUE THREAD

**Issue by jimrollenhagen** [meteor-gazelle/meteor-gazelle#163]
When interesting things happen in software, those things should be logged. Operators should be able to find out why things are breaking. So on and so forth. We should do this in a common way that allows operators to configure where logs are sent (console, file, logstash, mongo, etc).

Let's NOT use observatory[0], which is kind of a standard in the Meteor world. It logs to mongo, as far as I can tell, and has a built in GUI. We should be logging enough info that it would be insane to send to mongo. :)

A quick google found a blog post[1] that is using winston[2] for logging. Seems pretty legit to me.

We should probably have a small wrapper around winston so we can swap it out if we decide to do so. And for configuration, etc.

Thoughts?

[0] http://observatory.meteor.com/
[1] https://meteorhacks.com/logging-support-for-meteor
[2] https://github.com/winstonjs/winston

**Comment by ghost** [comment]
@jimrollenhagen Good ideas. Winston appears to be more flexible than Observatory and I agree on the Mongo point. Winston can handle storing data to Mongo if it's something we deem necessary. 

Any ideas on what we would be logging?

**Comment by c-p-b** [comment]
+1 to Winston.

Just as an example, when we send out certain HTTP responses that indicate uncaught exceptions we should be logging them for dev reference.

Behavior that intends to compromise the integrity of the system should also be logged, e.g. if a parameterized query exceeds a certain amount of characters as an input.

**Comment by rhomes-wcd** [comment]
+1 on Winston, and great idea for the having a wrapper. If we can decide on this I would love to utilize it in the ip bans enhancement.

**Comment by jimrollenhagen** [comment]
tl;dr log anything that an operator should know about. Different levels so ops that don't care as much can turn off the more verbose stuff.

**Comment by ghost** [comment]
@jimrollenhagen Do you have time to start writing the code for this? If so that would be great.
